### PR TITLE
ignore FQS graphql errors when there is also data returned

### PR DESCRIPTION
**Pull request by sarahmangels**
Fixes both https://zeushealth.atlassian.net/browse/CDEV-246 and https://zeushealth.atlassian.net/browse/CDEV-247

FQS can return both `errors` and `data` for a single request. 
This PR allows us to still use `data`, if it exists, even if there are `errors`.

 i got all the requests except these which are a bit different:
* https://github.com/zeus-health/ctw-component-library/blob/29b24fe333ed1c375f33e6dcdbb5f03f2c096f9f/src/components/content/resource/history.ts#L196
* https://github.com/zeus-health/ctw-component-library/blob/29b24fe333ed1c375f33e6dcdbb5f03f2c096f9f/src/components/content/resource/history.ts#L223
* https://github.com/zeus-health/ctw-component-library/blob/29b24fe333ed1c375f33e6dcdbb5f03f2c096f9f/src/services/fqs/long-poll-fqs.ts#L40
* https://github.com/zeus-health/ctw-component-library/blob/29b24fe333ed1c375f33e6dcdbb5f03f2c096f9f/src/services/fqs/queries/provenances.ts#L18

### CODE REVIEW

**Review by pmadria3 — APPROVED** [review]

**Review by pmadria3 — CHANGES_REQUESTED** [review]
Wait so those requests do not work with the new changes or?

**Review by pmadria3 — APPROVED** [review]

**Comment by sarahmangels** [comment]
> Wait so those requests do not work with the new changes or?

No they work the same as they always did I just didnt update them to be able to handle if the FQS query returned both errors and data. In that case those queries I didn't touch would just error out like they do right now.

We can update those ones in a separate PR

File: src/services/fqs/client.ts [hunk 1]
```diff
@@ -21,12 +23,28 @@ export interface GenericConnection<T extends ResourceTypeString> {
...
 };
 
+export const fqsRequest = async <T>(client: GraphQLClient, query: string, variables: object) => {
+  const { data, errors } = await client.rawRequest<T>(query, variables as Variables);
+  const fhirData = graphQLToFHIR(data);
+  if (errors) {
+    if (data) {
+      errors.forEach((e) => {
+        Telemetry.logger.error("Error in FQS request", undefined, e);
+      });
```
Comment (WesleyKapow) [inline src/services/fqs/client.ts:40]: Can we log just once for all of the errors?

**Review by WesleyKapow — APPROVED** [review]
LGTM

File: src/services/fqs/client.ts (hunk 1 above)
Comment (sarahmangels) [reply src/services/fqs/client.ts:40]: hmm I dont see a way to do that with Telemetry.logger.error but maybe im missing something

**Pull request merged**